### PR TITLE
Feature add org type and mysql

### PR DESCRIPTION
**Pull request by evenos**
- MySQL connector
- Added SÆR as a public organizationtype

### CODE REVIEW

**Comment by nuth** [comment]
SonarQube analysis reported 99 issues
* CRITICAL 14 critical
* MAJOR 79 major
* MINOR 6 minor

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. CRITICAL no.difi.meldingsutveksling:serviceregistry-parent: Filename: mysql-connector-java-5.1.40.jar | Reference: CVE-2012-0882 | CVSS Score: 7.5 | Category: CWE-119 Improper Restriction of Operations within the Bounds of a Memory Buffer | Buffer overflow in yaSSL, as used in MySQL 5.5.20 and possibly other versions including 5.5.x before 5.5.22 and 5.1.x before 5.1.62, allows remote attackers to execute arbitrary code via unspecified vectors, as demonstrated by VulnDisco Pack Professional 9.17.  NOTE: as of 20120224, this disclosure has no actionable information. However, because the module author is a reliable researcher, the issue is being assigned a CVE identifier for tracking purposes.  NOTE: due to lack of details, it is not clear whether this issue is a duplicate of CVE-2012-0492 or another CVE. [rule](http://sonar-app01.difi.local:9000/coding_rules#rule_key=OWASP%3AUsingComponentWithKnownVulnerability)
1. CRITICAL no.difi.meldingsutveksling:serviceregistry-parent: Filename: mysql-connector-java-5.1.40.jar | Reference: CVE-2012-0553 | CVSS Score: 7.5 | Category: CWE-119 Improper Restriction of Operations within the Bounds of a Memory Buffer | Buffer overflow in yaSSL, as used in MySQL 5.1.x before 5.1.68 and 5.5.x before 5.5.28, has unspecified impact and attack vectors, a different vulnerability than CVE-2013-1492. [rule](http://sonar-app01.difi.local:9000/coding_rules#rule_key=OWASP%3AUsingComponentWithKnownVulnerability)
1. CRITICAL no.difi.meldingsutveksling:serviceregistry-parent: Filename: mysql-connector-java-5.1.40.jar | Reference: CVE-2008-0226 | CVSS Score: 7.5 | Category: CWE-119 Improper Restriction of Operations within the Bounds of a Memory Buffer | Multiple buffer overflows in yaSSL 1.7.5 and earlier, as used in MySQL and possibly other products, allow remote attackers to execute arbitrary code via (1) the ProcessOldClientHello function in handshake.cpp or (2) "input_buffer& operator>>" in yassl_imp.cpp. [rule](http://sonar-app01.difi.local:9000/coding_rules#rule_key=OWASP%3AUsingComponentWithKnownVulnerability)
1. CRITICAL no.difi.meldingsutveksling:serviceregistry-parent: Filename: commons-collections4-4.1.jar | Reference: CVE-2015-6420 | CVSS Score: 7.5 | Serialized-object interfaces in certain Cisco Collaboration and Social Media; Endpoint Clients and Client Software; Network Application, Service, and Acceleration; Network and Content Security Devices; Network Management and Provisioning; Routing and Switching - Enterprise and Service Provider; Unified Computing; Voice and Unified Communications Devices; Video, Streaming, TelePresence, and Transcoding Devices; Wireless; and Cisco Hosted Services products allow remote attackers to execute arbitrary commands via a crafted serialized Java object, related to the Apache Commons Collections (ACC) library. [rule](http://sonar-app01.difi.local:9000/coding_rules#rule_key=OWASP%3AUsingComponentWithKnownVulnerability)
1. CRITICAL no.difi.meldingsutveksling:serviceregistry-parent: Filename: tomcat-embed-core-8.5.6.jar | Reference: CVE-2017-5651 | CVSS Score: 7.5 | Category: CWE-19 Data Handling | In Apache Tomcat 9.0.0.M1 to 9.0.0.M18 and 8.5.0 to 8.5.12, the refactoring of the HTTP connectors introduced a regression in the send file processing. If the send file processing completed quickly, it was possible for the Processor to be added to the processor cache twice. This could result in the same Processor being used for multiple requests which in turn could lead to unexpected errors and/or response mix-up. [rule](http://sonar-app01.difi.local:9000/coding_rules#rule_key=OWASP%3AUsingComponentWithKnownVulnerability)
1. CRITICAL no.difi.meldingsutveksling:serviceregistry-parent: Filename: tomcat-embed-core-8.5.6.jar | Reference: CVE-2016-8735 | CVSS Score: 7.5 | Category: CWE-284 Improper Access Control | Remote code execution is possible with Apache Tomcat before 6.0.48, 7.x before 7.0.73, 8.x before 8.0.39, 8.5.x before 8.5.7, and 9.x before 9.0.0.M12 if JmxRemoteLifecycleListener is used and an attacker can reach JMX ports. The issue exists because this listener wasn't updated for consistency with the CVE-2016-3427 Oracle patch that affected credential types. [rule](http://sonar-app01.difi.local:9000/coding_rules#rule_key=OWASP%3AUsingComponentWithKnownVulnerability)
1. CRITICAL no.difi.meldingsutveksling:serviceregistry-parent: Filename: tomcat-embed-core-8.5.6.jar | Reference: CVE-2016-6325 | CVSS Score: 7.2 | Category: CWE-264 Permissions, Privileges, and Access Controls | The Tomcat package on Red Hat Enterprise Linux (RHEL) 5 through 7, JBoss Web Server 3.0, and JBoss EWS 2 uses weak permissions for (1) /etc/sysconfig/tomcat and (2) /etc/tomcat/tomcat.conf, which allows local users to gain privileges by leveraging membership in the tomcat group. [rule](http://sonar-app01.difi.local:9000/coding_rules#rule_key=OWASP%3AUsingComponentWithKnownVulnerability)
1. CRITICAL no.difi.meldingsutveksling:serviceregistry-parent: Filename: tomcat-embed-core-8.5.6.jar | Reference: CVE-2016-5425 | CVSS Score: 7.2 | Category: CWE-264 Permissions, Privileges, and Access Controls | The Tomcat package on Red Hat Enterprise Linux (RHEL) 7, Fedora, CentOS, Oracle Linux, and possibly other Linux distributions uses weak permissions for /usr/lib/tmpfiles.d/tomcat.conf, which allows local users to gain root privileges by leveraging membership in the tomcat group. [rule](http://sonar-app01.difi.local:9000/coding_rules#rule_key=OWASP%3AUsingComponentWithKnownVulnerability)
1. CRITICAL no.difi.meldingsutveksling:serviceregistry-parent: Filename: tomcat-embed-websocket-8.5.6.jar | Reference: CVE-2017-5651 | CVSS Score: 7.5 | Category: CWE-19 Data Handling | In Apache Tomcat 9.0.0.M1 to 9.0.0.M18 and 8.5.0 to 8.5.12, the refactoring of the HTTP connectors introduced a regression in the send file processing. If the send file processing completed quickly, it was possible for the Processor to be added to the processor cache twice. This could result in the same Processor being used for multiple requests which in turn could lead to unexpected errors and/or response mix-up. [rule](http://sonar-app01.difi.local:9000/coding_rules#rule_key=OWASP%3AUsingComponentWithKnownVulnerability)
1. CRITICAL no.difi.meldingsutveksling:serviceregistry-parent: Filename: tomcat-embed-websocket-8.5.6.jar | Reference: CVE-2016-8735 | CVSS Score: 7.5 | Category: CWE-284 Improper Access Control | Remote code execution is possible with Apache Tomcat before 6.0.48, 7.x before 7.0.73, 8.x before 8.0.39, 8.5.x before 8.5.7, and 9.x before 9.0.0.M12 if JmxRemoteLifecycleListener is used and an attacker can reach JMX ports. The issue exists because this listener wasn't updated for consistency with the CVE-2016-3427 Oracle patch that affected credential types. [rule](http://sonar-app01.difi.local:9000/coding_rules#rule_key=OWASP%3AUsingComponentWithKnownVulnerability)

**Comment by nuth** [comment]
SonarQube analysis reported 4 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR no.difi.meldingsutveksling:serviceregistry-parent: Filename: spring-cloud-starter-config-1.1.0.RELEASE.jar | Reference: CVE-2016-9878 | CVSS Score: 5.0 | Category: CWE-22 Improper Limitation of a Pathname to a Restricted Directory ('Path Traversal') | An issue was discovered in Pivotal Spring Framework before 3.2.18, 4.2.x before 4.2.9, and 4.3.x before 4.3.5. Paths provided to the ResourceServlet were not properly sanitized and as a result exposed to directory traversal attacks. [rule](http://sonar-app01.difi.local:9000/coding_rules#rule_key=OWASP%3AUsingComponentWithKnownVulnerability)
1. MAJOR no.difi.meldingsutveksling:serviceregistry-parent: Filename: spring-cloud-starter-eureka-1.1.0.RELEASE.jar | Reference: CVE-2016-9878 | CVSS Score: 5.0 | Category: CWE-22 Improper Limitation of a Pathname to a Restricted Directory ('Path Traversal') | An issue was discovered in Pivotal Spring Framework before 3.2.18, 4.2.x before 4.2.9, and 4.3.x before 4.3.5. Paths provided to the ResourceServlet were not properly sanitized and as a result exposed to directory traversal attacks. [rule](http://sonar-app01.difi.local:9000/coding_rules#rule_key=OWASP%3AUsingComponentWithKnownVulnerability)
1. MAJOR no.difi.meldingsutveksling:serviceregistry-parent: Filename: spring-cloud-starter-ribbon-1.1.0.RELEASE.jar | Reference: CVE-2016-9878 | CVSS Score: 5.0 | Category: CWE-22 Improper Limitation of a Pathname to a Restricted Directory ('Path Traversal') | An issue was discovered in Pivotal Spring Framework before 3.2.18, 4.2.x before 4.2.9, and 4.3.x before 4.3.5. Paths provided to the ResourceServlet were not properly sanitized and as a result exposed to directory traversal attacks. [rule](http://sonar-app01.difi.local:9000/coding_rules#rule_key=OWASP%3AUsingComponentWithKnownVulnerability)
1. MAJOR no.difi.meldingsutveksling:serviceregistry-parent: Filename: spring-cloud-starter-1.1.0.RELEASE.jar | Reference: CVE-2016-9878 | CVSS Score: 5.0 | Category: CWE-22 Improper Limitation of a Pathname to a Restricted Directory ('Path Traversal') | An issue was discovered in Pivotal Spring Framework before 3.2.18, 4.2.x before 4.2.9, and 4.3.x before 4.3.5. Paths provided to the ResourceServlet were not properly sanitized and as a result exposed to directory traversal attacks. [rule](http://sonar-app01.difi.local:9000/coding_rules#rule_key=OWASP%3AUsingComponentWithKnownVulnerability)

File: serviceregistry-server/src/main/java/no/difi/meldingsutveksling/serviceregistry/model/OrganizationTypes.java [hunk 1]
```diff
@@ -35,6 +37,6 @@
      * @return collection of all known public organization types
      */
     public Set<OrganizationType> publicOrganization() {
-        return Sets.newHashSet(ORGL);
+        return Sets.newHashSet(ORGL, SAR);
     }
 }
```
Comment (martinwam) [inline serviceregistry-server/src/main/java/no/difi/meldingsutveksling/serviceregistry/model/OrganizationTypes.java]: Kan revertere disse endringene for å unngå mergekonflikt. Klassen fjernes.

File: pom.xml [hunk 2]
```diff
@@ -160,6 +160,11 @@
                 <artifactId>lombok</artifactId>
                 <version>1.16.10</version>
             </dependency>
+            <dependency>
+                <groupId>mysql</groupId>
+                <artifactId>mysql-connector-java</artifactId>
+                <version>6.0.6</version>
+            </dependency>
```
Comment (martinwam) [inline pom.xml]: Trenger ikke definere denne, da versjonen kommer fra spring-boot parent, som igjen kommer  fra move-dependencies.

**Comment by evenos** [comment]
feature removed instead of changed in PR-120 MOVE-263

**Pull request closed without merging**